packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -33,10 +34,15 @@ export class MetadataOutlineProvider
     this.defaultOrg = usernameOrAlias;
   }
 
-  public async refresh(): Promise<void> {
-    const usernameOrAlias = await this.getDefaultUsernameOrAlias();
-    this.defaultOrg = usernameOrAlias;
-    this.internalOnDidChangeTreeData.fire();
+  public async refresh(node?: BrowserNode): Promise<void> {
+    if (node && !node.toRefresh) {
+      node.toRefresh = true;
+    } else if (!this.toRefresh) {
```

Comment:
Is this check necessary? It looks like the refresh function should only get called if the user is attempting to refresh- and if it's not a refresh at the type level, it has to be a refresh at the org level. Can we make this an `else`?

Reply:
The condition is to stop people from spamming refresh and making a ton of api calls if one is already in progress.